### PR TITLE
fix: selects default endpoint randomly

### PR DESCRIPTION
**Pull request by impelcrypto**
**Pull Request Summary**

* fix: selects default endpoint randomly
  * currently, the Astar portal connects to the WSS endpoint maintained by the Astar foundation as default. This PR changes it to connect randomly for first-time visitors 

**Check list**

- [ ] contains breaking changes
- [ ] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Changes**
Before:
The portal connects to wss://rpc.astar.network automatically for first-time visitors.

After:
The portal connects to endpoints by randomly selecting endpoints.

Notes:
For those who have visited the Astar portal already (with `selectedEndpoint: wss://rpc.astar.network` in the browser’s local storage), I’ll temporarily set a one-time logic to remove the selected endpoint from local storage and the portal selects the endpoint randomly. This logic will be removed 1 month after the portal has been updated.

### CODE REVIEW

File: tests/dashboard.spec.ts [hunk 1]
```diff
@@ -8,4 +10,18 @@ test.describe('on dashboard screen', () => {
...
   });
+  test('Endpoint has been selected randomly', async ({ page }) => {
+    const isAppliedRandomEndpoint = await page.evaluate(() => {
+      return localStorage.getItem('isAppliedRandomEndpoint');
+    });
+    const selectedEndpointObj: string = (await page.evaluate(() => {
+      return localStorage.getItem('selectedEndpoint');
+    })) as string;
+    const selectedEndpoint = JSON.parse(selectedEndpointObj)[0];
+    const isSomeOfAstarEndpoints = providerEndpoints[endpointKey.ASTAR].endpoints.some(
```
Comment (bobo-k2) [inline tests/dashboard.spec.ts:21]: Maybe to add check that selected endpoint is not light client

**Review by bobo-k2 — APPROVED** [review]
LGTM

File: tests/dashboard.spec.ts (hunk 1 above)
Comment (impelcrypto) [reply tests/dashboard.spec.ts:22]: Ok, I've updated test in [this](https://github.com/AstarNetwork/astar-apps/pull/820/commits/4865bfb7d5d7da7c1975a65be25850ae6aa464b5) commit.

File: tests/dashboard.spec.ts [hunk 2]
```diff
@@ -21,7 +22,9 @@ test.describe('on dashboard screen', () => {
     const isSomeOfAstarEndpoints = providerEndpoints[endpointKey.ASTAR].endpoints.some(
       (it) => it.endpoint === selectedEndpoint
     );
+    const isNotLightClient = checkIsLightClient(selectedEndpoint);
```
Comment (bobo-k2) [inline tests/dashboard.spec.ts:25]: Looks like Not is not needed in the variable name. If you name variable like that I will expect the following condition `expect(isNotLightClient).toBe(true);`
just rename the variable to `isLightClient` and everything will be fine :)

Comment (impelcrypto) [reply tests/dashboard.spec.ts]: Good catch! I've fixed it in [this](https://github.com/AstarNetwork/astar-apps/pull/820/commits/2aba3fccdc27e75fb4e818d48c26c037e1d421c8) commit.

File: src/hooks/useAppRouter.ts [hunk 3]
```diff
@@ -55,16 +57,23 @@ export function useAppRouter() {
...
+
     const isReload =
       networkIdxStore === null ||
       networkIdxStore === undefined ||
       storedNetworkAlias !== networkParam;
     if (isReload) {
-      // Memo: `defaultCurrency` and `selectedEndpoint` are updated when the app connects to the API
-      const isFirstTimeVisitor = localStorage.length === 2;
+      // Memo: `defaultCurrency`, `isAppliedRandomEndpoint` and `selectedEndpoint` are updated when the app connects to the API
+      const isFirstTimeVisitor = localStorage.length === 3;
```
Comment (sirius651) [inline src/hooks/useAppRouter.ts:73]: why don't we check each key of local storage? If we manipulate local storage elsewhere, there is a high possibility of a bug.

**Review by sirius651 — CHANGES_REQUESTED** [review]

File: src/hooks/useAppRouter.ts (hunk 3 above)
Comment (impelcrypto) [reply src/hooks/useAppRouter.ts]: @sirius651  Fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/820/commits/34bba494effeb449c47f84856b563c6a7ea6b9f2) commit.

**Review by sirius651 — APPROVED** [review]

**Pull request merged**